include/boost/math/tools/color_maps.hpp

```diff
@@ -0,0 +1,156 @@
+//  (C) Copyright Nick Thompson 2021.
+//  (C) Copyright Matt Borland 2022.
+//  Use, modification and distribution are subject to the
+//  Boost Software License, Version 1.0. (See accompanying file
+//  LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+
+#ifndef BOOST_MATH_COLOR_MAPS_HPP
+#define BOOST_MATH_COLOR_MAPS_HPP
+
+#include <cmath>
+#include <array>
+#include <algorithm>
+#include <utility>
+#include <boost/math/tools/concepts.hpp>
+#include <boost/math/ccmath/floor.hpp>
+
+namespace boost::math::tools {
+
+namespace detail {
+
+template <BOOST_MATH_FLOATING_POINT_TYPE Real>
+class color_map
+{
+private:
+    std::array<Real, 256> r;
+    std::array<Real, 256> g;
```

Comment:
One thing I worry about with this design is that there is no fundamental reason that users should be required to choose an array of length 256. You can see I simply chose one length from the [original reference](https://www.kennethmoreland.com/color-advice/), but the reference itself specifies that there is an analytic form to many of these which indicates that it might be very reasonable for a user to specify basically any continuous function from [0,1]->[0,1]^3.. .

Reply:
I changed the base class to take a `std::size_t` instead of constraining ourselves to 256. The `custom_color_map` can now be specified by for example `custom_color_map<double, 512>(r, g, b)`, but our provided ones are not affected.